lib/rouge/lexers/apex.rb

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class Apex < RegexLexer
+      title "Apex"
+      desc "The Apex programming language (provided by salesforce)"
+
+      tag 'apex'
+      filenames '*.cls'
+      mimetypes 'text/x-apex'
+
+      keywords = %w(
+        assert break case catch continue default do else finally for
+        if goto instanceof new return switch this throw try while insert
+        update delete
+      )
+
+      declarations = %w(
+        abstract const enum extends final implements native private protected
+        public static super synchronized throws transient volatile with
+        sharing without inherited virtual global testmethod
+      )
+
+      soql = %w(
+        SELECT FROM WHERE  UPDATE LIKE TYPEOF END USING  SCOPE WITH DATA
+        CATEGORY GROUP BY ROLLUP CUBE HAVING  ORDER BY ASC DESC NULLS FIRST LAST
+        LIMIT OFFSET FOR VIEW REFERENCE UPDATE TRACKING VIEWSTAT OR AND
+      )
+
+      types = %w(String boolean byte char double float int long short var void)
+
+      id = /[a-zA-Z_][a-zA-Z0-9_]*/
+
+      state :root do
+        rule /[^\S\n]+/, Text
+        rule %r(//.*?$), Comment::Single
+        rule %r(/\*.*?\*/)m, Comment::Multiline
+        rule /(?:#{keywords.join('|')})\b/, Keyword
+        rule /(?:#{soql.join('|')})\b/, Keyword
+
+        rule %r(
+          (\s*(?:[a-zA-Z_][a-zA-Z0-9_.\[\]<>]*\s+)+?) # return arguments
+          ([a-zA-Z_][a-zA-Z0-9_]*)                  # method name
+          (\s*)(\()                                 # signature start
+        )mx do |m|
+          delegate Apex, m[1]
```

Comment:
Delegation is intended for languages that embed another language within them (think how JavaScript can be inside HTML). What are you trying to do here?